### PR TITLE
Rename other occs of (Code/File)Map to Source(Map/File) #51574

### PR DESCRIPTION
**Pull request by dlavati**
Additional renamings for #51574.

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -1164,7 +1164,7 @@ Compatibility Notes
 - [Changes in method matching against higher-ranked types][43880] This may cause
   breakage in subtyping corner cases. [A more in-depth explanation is available.][info/43880]
 - [rustc's JSON error output's byte position start at top of file.][42973]
-  Was previously relative to the rustc's internal `CodeMap` struct which
+  Was previously relative to the rustc's internal `SourceMap` struct which
```
Comment (bjorn3) [inline RELEASES.md]: I don't think this sould be changed, because `CodeMap` was the name back then.

**Review by bjorn3 — CHANGES_REQUESTED** [review]

File: RELEASES.md (hunk 1 above)
Comment (dlavati) [reply RELEASES.md]: You're right, it does speak in past tense.

**Comment by bjorn3** [comment]
@dlavati i'm not part of the rust lang org. :)

**Comment by dlavati** [comment]
r?

**Comment by petrochenkov** [comment]
Could you keep the naming scheme consistent with https://doc.rust-lang.org, i.e.

`caching_sourcemap_view` -> `caching_source_map_view`
`StableSourcefileId` -> `StableSourceFileId`
`MalformedForSourcemap` -> `MalformedForSourceMap`
`MalformedSourcemapPositions` -> `MalformedSourceMapPositions`

, and also squash all the changes into one commit.

**Comment by petrochenkov** [comment]
@dlavati 
This still needs a rebase.

**Pull request merged**